modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinClientCodegen.java

```diff
@@ -438,6 +441,10 @@ public void processOpts() {
             setGenerateOneOfAnyOfWrappers(Boolean.parseBoolean(additionalProperties.get(GENERATE_ONEOF_ANYOF_WRAPPERS).toString()));
         }
 
+        if (!additionalProperties.containsKey(FAIL_ON_UNKNOWN_PROPERTIES)) {
+            additionalProperties.put(FAIL_ON_UNKNOWN_PROPERTIES, false);
+        }
```

Comment:
does this allow setting `FAIL_ON_UNKNOWN_PROPERTIES` to true?

Reply:
As far as I understand the generator, `FAIL_ON_UNKNOWN_PROPERTIES` will be true if and only if the we specify the `failOnUnknownProperties` option to true. 

Am I right?

Reply:
I'm new to the repository, so I may be wrong here, but reading these lines, it looks as though we only update the property value (and to _false_) if the property isn't included by the user configuration.

The other properties are in this format: 

```
        if (additionalProperties.containsKey(KEY)) {
            setKey(additionalProperties.get(KEY).toString());
        }
```

 such as:

```
        if (additionalProperties.containsKey(REQUEST_DATE_CONVERTER)) {
            setRequestDateConverter(additionalProperties.get(REQUEST_DATE_CONVERTER).toString());
        }
```

Which reads as though - if provided, we will override the value based on the provided value. I'm sure a unit test will make this clear, though.

Reply:
Indeed, you are right, thanks!

I refactored this part and  added a unit test for this.